### PR TITLE
ft #155931587: Accesing the Api

### PR DESCRIPTION
**Pull request by King-Benx**
#### What does this PR do?
This PR delivers accessing to the API ie the GithubApi

#### Description of Task to be completed?
This task delivers access to the API, the GithubPresenter.java leverages on the GithubApi interface.

#### SCREENSHOT
screen shot 2018-08-13 at 11 48 16

#### How should this be manually tested?
1. Clone the repository onto your machine and build the app. 
2. Inside the src/main/java/com.example.asiimwebenard.javadevelopersinnairobi/com.example.asiimwebenard.javadevelopersinnairobi.presenter there is the GithubPresenter.java class that leverages on the GithubApi.java class

#### Background context
NONE

#### What are the relevant pivotal tracker stories?
[Delivers #155931587]

### CODE REVIEW

**Comment by Jak-Sidious** [comment]
At this point you are supposed to be using actual data from the API and not your dummy data

**Comment by King-Benx** [comment]
Actual data being used is in another story of day 7

File: app/src/main/java/com/example/asiimwebenard/javadevelopersinnairobi/presenter/GithubPresenter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    public GithubPresenter(GithubUserView githubUserView) {
+        this.githubUserView = githubUserView;
+        if (this.githubService == null){
+            this.githubService = new GithubService();
+        }
+    }
+
+    public void getAllUsers(){
+        githubService.githubApi().getAllUsers().enqueue(new Callback<Data>() {
```
Comment (Jak-Sidious) [inline app/src/main/java/com/example/asiimwebenard/javadevelopersinnairobi/presenter/GithubPresenter.java]: Granted this is acceptable, I do believe it is part of the style guide to put each individual function that is preceded by a . like `.githubApi()` should be on it's individual line simillar to how you declared them in the `GithubService.java` file

**Comment by Jak-Sidious** [comment]
LGTM

**Pull request merged**